src/main/java/de/tum/in/www1/artemis/web/rest/ModelingSubmissionResource.java

```diff
@@ -278,8 +280,70 @@ public ModelingSubmissionResource(ModelingSubmissionService modelingSubmissionSe
     }
 
     /**
-     * Removes sensitive information (e.g. example solution) from the exercise. This should be called before sending an exercise to the client for a student. IMPORTANT: Do not call
-     * this method from a transactional context as this would remove the sensitive information also from the entity in the database without explicitly saving it.
+     * Returns the submission with data needed for the modeling editor, which includes the participation, the model and the result (if the assessment was already submitted).
+     *
+     * @param participationId the participationId for which to find the data for the modeling editor
+     * @return the ResponseEntity with json as body
+     */
+    @GetMapping("/modeling-editor/{participationId}")
+    @PreAuthorize("hasAnyRole('USER', 'TA', 'INSTRUCTOR', 'ADMIN')")
+    @Transactional(readOnly = true)
+    public ResponseEntity<ModelingSubmission> getSubmissionForModelingEditor(@PathVariable Long participationId) {
+        Participation participation = participationService.findOneWithEagerSubmissionsAndResults(participationId);
+        if (participation == null) {
+            return ResponseEntity.badRequest()
+                    .headers(HeaderUtil.createFailureAlert(applicationName, true, ENTITY_NAME, "participationNotFound", "No participation was found for the given ID.")).body(null);
+        }
+        ModelingExercise modelingExercise;
+        if (participation.getExercise() instanceof ModelingExercise) {
+            modelingExercise = (ModelingExercise) participation.getExercise();
+            if (modelingExercise == null) {
+                return ResponseEntity.badRequest()
+                        .headers(HeaderUtil.createFailureAlert(applicationName, true, "modelingExercise", "exerciseEmpty", "The exercise belonging to the participation is null."))
+                        .body(null);
+            }
+
+            // make sure sensitive information are not sent to the client
+            modelingExercise.filterSensitiveInformation();
+        }
+        else {
+            return ResponseEntity.badRequest().headers(
+                    HeaderUtil.createFailureAlert(applicationName, true, "modelingExercise", "wrongExerciseType", "The exercise of the participation is not a modeling exercise."))
+                    .body(null);
+        }
+
+        // Students can only see their own models (to prevent cheating). TAs, instructors and admins can see all models.
+        if (!(authCheckService.isOwnerOfParticipation(participation) || courseService.userHasAtLeastTAPermissions(modelingExercise.getCourse()))) {
```

Comment:
Can we also use `authCheckService.isAtLeastTeachingAssisstantInCourse(modelingExercise.getCourse())`?
I think it would be better if we concentrate this kind of permission checking in one service.

Reply:
I changed it to use `authCheckService.isAtLeastTeachingAssistantForExercise(modelingExercise)`.